forwarder/kafka/kafka_helpers.py

```diff
@@ -11,16 +11,50 @@
 
 from .kafka_producer import KafkaProducer
 
+DEFAULT_SASL_MECHANISM = "SCRAM-SHA-256"
+
+
+def get_sasl_config(
+    mechanism: str, username: Optional[str] = None, password: Optional[str] = None
+) -> dict:
+    """Return a dict with SASL configuration parameters.
+    Supported protocols: SASL_PLAINTEXT (i.e. without TLS)
+    Supported mechanisms: PLAIN, SCRAM-SHA-512, SCRAM-SHA-256.
+    Note that whereas some SASL mechanisms do not require user/password, the three
+    we currently support do.
+    """
+    supported_sasl_mechanisms = ["PLAIN", "SCRAM-SHA-512", "SCRAM-SHA-256"]
+    if mechanism not in supported_sasl_mechanisms:
+        raise RuntimeError(
+            f"SASL mechanism {mechanism} not supported, use one of {supported_sasl_mechanisms}"
+        )
+    if not username or not password:
+        raise RuntimeError(
+            f"Username and password must be provided to use SASL {mechanism}"
+        )
+    sasl_config = {
+        "sasl.mechanism": mechanism,
+        "security.protocol": "SASL_PLAINTEXT",  # SASL_PLAINTEXT for plaintext, SASL_SSL for encrypted
+        "sasl.username": username,
+        "sasl.password": password,
+    }
+    return sasl_config
+
 
 def create_producer(
     broker_address: str,
+    sasl_mechanism: Optional[str] = None,
```

Comment:
In discussions with Matt we think that we should perhaps build the kafka configuration (incl. SASL) before calling create_producer / create_consumer, and then pass a single config dictionary to those functions.

We decide to leave it like this for now because it is simple enough and we may need to refactor a few places if we do that change.
But we may consider rewriting it if we keep doing changes to this code.